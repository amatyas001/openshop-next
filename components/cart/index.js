export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
