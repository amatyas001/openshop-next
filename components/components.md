Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
