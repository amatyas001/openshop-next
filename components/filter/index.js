export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
