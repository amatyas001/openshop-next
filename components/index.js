export * from './shared';
export * from './cart';
export * from './payment';
