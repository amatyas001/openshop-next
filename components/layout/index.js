export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
