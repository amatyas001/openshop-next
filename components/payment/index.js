export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
