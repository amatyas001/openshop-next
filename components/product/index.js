export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
