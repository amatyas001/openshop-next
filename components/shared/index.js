export * from './Button/Button';
export * from './Card/Card';
export * from './Field/Field';
export * from './Rating/Rating';
export * from './ScrollTop/ScrollTop';
export * from './Navbar/Navbar';
export * from './Footer/Footer';
export * from './Spinner/Spinner';
