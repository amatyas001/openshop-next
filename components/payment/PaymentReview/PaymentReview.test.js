import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import { paymentForm } from '@app/redux/actions';
import { PaymentReview } from '@app/components';

const mockStore = createStore([]);

describe('<PaymentReview />', () => {
  let tree,
    back = jest.fn(),
    store = mockStore({
      cart: [
        {
          id: 'mock_id',
          name: 'mock_name',
          desc: 'mock_desc',
          img: 'mock_img',
          price: 10,
          rating: 5,
        },
      ],
    });

  beforeAll(() => {
    jest
      .spyOn(require('next/router'), 'useRouter')
      .mockImplementationOnce(() => ({
        back,
      }));

    act(() => {
      tree = create(
        <Provider store={store}>
          <PaymentReview />
        </Provider>
      );
    });

    store.clearActions();
  });

  it('should render without props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should push back router', () => {
    act(() => {
      tree.root
        .findByProps({ 'data-testid': 'review-button-back' })
        .props.onClick();
    });

    expect(back).toHaveBeenCalledTimes(1);
  });

  it('should push form state', () => {
    act(() => {
      tree.root
        .findByProps({ 'data-testid': 'review-button-form' })
        .props.onClick();
    });

    expect(store.getActions()).toEqual([paymentForm()]);
  });
});
