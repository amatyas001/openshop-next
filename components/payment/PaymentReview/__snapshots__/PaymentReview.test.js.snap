// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentReview /> should render without props 1`] = `
<div
  className="css-dvxtzn"
>
  <section
    className="css-8atqhb"
  >
    <ul
      className="css-0"
      role="list"
    >
      <li
        className="css-tic8xi"
        role="listitem"
      >
        <img
          alt="mock_name"
          className="css-rhemcv"
        />
        <div
          className="css-1z0sfv6"
        >
          <strong
            className="css-18poma1"
          >
            mock_name
          </strong>
          <em
            className="css-183x3ib"
          />
        </div>
        <p
          className="css-15x91gx"
        >
          mock_desc
        </p>
        <strong
          className="css-nwzge5"
        >
          10.00
           $
        </strong>
        <div
          aria-label="Delete"
          className="css-c4ibz5"
          data-cy="button"
          href="#void"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            />
          </svg>
        </div>
      </li>
    </ul>
    <footer
      className="css-0"
    >
      <p
        className="css-ohvq3n"
      >
        TOTAL PRICE
      </p>
      <strong
        className="css-lk14dv"
      >
         $
      </strong>
    </footer>
  </section>
  <div
    className="css-1hojse"
  >
    <button
      className="css-1h13z07"
      data-cy="button"
      data-testid="review-button-back"
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      BACK TO SHOPPING
    </button>
    <button
      className="css-187kagv"
      data-cy="button"
      data-testid="review-button-form"
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      PAYMENT DETAILS
    </button>
  </div>
</div>
`;
